be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -369,6 +370,21 @@ final case class DbActor(
     (publicKey, privateKey)
   }
 
+  @throws[DbActorNAckException]
+  private def generateHeartbeat(): Option[HashMap[Channel, Set[Hash]]] = {
```

Comment:
I feel like an `Option` is not necessary here, you might as well return an empty map directly ?